### DISCUSSION TITLE
Could not query double values after out() API

### DISCUSSION THREAD

**Discussion by zhengyingying** [JanusGraph/janusgraph#2754] [Q&A] [answered]
Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 0.5.3
- Storage Backend: inmemory
- Mixed Index Backend: none
- Link to discussed bug: 
- Expected Behavior:
When we execute `g.V().out('el0').has('vp2', 3.49203472E8)`, we want to get the required vertices.
- Current Behavior:
Now, an exception `java.lang.NumberFormatException` was throw. We found that when we add the type DOUBLE explicitly, i.e.,  `g.V().out('el0').has('vp2', 3.49203472E8d)`, we can get the result. And we also can get result with the query statement `g.V().has('vp2', 3.49203472E8)`.
- Steps to Reproduce: 

### Stack Trace (if you have one)

```
java.util.concurrent.ExecutionException: org.apache.tinkerpop.gremlin.driver.exception.ResponseException: java.lang.NumberFormatException
```

**Comment by porunov** [comment]
Could you please add reproduction steps including schema creation and data loading?

**Comment by li-boxuan** [comment, marked as answer]
Did you do this on gremlin console? It seems groovy interprets 3.49203472E8 as `java.math.BigDecimal` which is not a supported data type in JanusGraph. To be safe, always explicitly use `3.49203472E8d`.

**Comment by zhengyingying** [comment]
@porunov We before met this problem, but not save data. I will provide the reproduction steps when we met this problem again. 
@li-boxuan Yes, I did this query on gremlin console.  It is strange that when we executed `g.V().has('vp2', 3.49203472E8)`, no exception was throw. Maybe it is not safe using `3.49203472E8`.

**Reply by porunov** [reply]
I converted this issue to Discussions. I believe @li-boxuan pointed on the root cause of such behavior. 
I don't think this is a bug on JanusGraph side but rather the way Gremlin Console works. You should explicitly define double values in Gremlin Console. In case you get this error with explicit values, please, let us know
